### PR TITLE
Upgrade webpack

### LINKED ISSUE #83: Upgrade webpack

### PR DESCRIPTION
**Pull request by SimenB**
Fixes #83

You should probably test this properly before merge.

A build works, and the dev server started, but I didn't start up the API, so haven't tested navigating

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -59,30 +59,29 @@
...
     "eslint": "^4.1.1",
     "eslint-config-finn": "^2.0.0",
     "eslint-config-finn-react": "^2.0.0",
     "eslint-plugin-react": "^7.1.0",
-    "extract-text-webpack-plugin": "^1.0.1",
+    "extract-text-webpack-plugin": "^3.0.0-rc.0",
     "identity-obj-proxy": "^3.0.0",
     "jest": "^20.0.4",
-    "node-sass": "~3.12.1",
-    "postcss-loader": "^0.13.0",
```
Comment (SimenB) [inline package.json]: unused

File: src/data/applications-api.js [hunk 2]
```diff
@@ -34,7 +34,7 @@ function storeApplicationMetaData (appName, key, value) {
     }).then(throwIfNotSuccess);
 }
 
-module.exports = {
```
Comment (SimenB) [inline src/data/applications-api.js]: all of these `import/export` changes were because webpack complained

File: .babelrc [hunk 3]
```diff
@@ -1,9 +1,12 @@
 {
-  "presets": ["react", "es2015", "stage-2"],
+  "presets": ["react", ["es2015", {"modules": false}], "stage-2"],
```
Comment (SimenB) [inline .babelrc]: might consider `preset-env`?

**Comment by SimenB** [comment]
@sveisvei did you test it?

Afraid we might hit https://github.com/webpack-contrib/extract-text-webpack-plugin/issues/548#issuecomment-313641414

File: package.json [hunk 4]
```diff
@@ -25,11 +25,11 @@
   "license": "Apache-2.0",
   "scripts": {
     "build": "rm -rf dist && mkdir -p dist/public && npm run build:assets && npm run build:html && npm run build:img && npm run build:ico",
-    "build:assets": "NODE_ENV=production webpack -p",
+    "build:assets": "NODE_ENV=production webpack -p --display-optimization-bailout",
     "build:html": "cp public/*.html dist/.",
     "build:ico": "cp public/*.ico dist/.",
     "build:img": "cp public/*.png dist/public/.",
-    "start": "NODE_ENV=development webpack-dev-server --config webpack.config.js --progress --colors",
+    "start": "NODE_ENV=development webpack-dev-server --progress --colors --display-optimization-bailout",
```
Comment (ivarconr) [inline package.json]: webpack-dev-server says `Unknown argument: display-optimization-bailout`!

Comment (SimenB) [inline package.json]: Ah, I didn't actually test that... It can be removed, it's really just useful for production

**Pull request merged**